pkg/storage/store_snapshot.go

```diff
@@ -139,14 +139,19 @@ func (kvSS *kvBatchSnapshotStrategy) Receive(
 				err = errors.Wrap(err, "invalid snapshot")
 				return IncomingSnapshot{}, sendSnapshotError(stream, err)
 			}
-
 			inSnap := IncomingSnapshot{
 				SnapUUID:   snapUUID,
 				Batches:    batches,
 				LogEntries: logEntries,
 				State:      &header.State,
 				snapType:   snapTypeRaft,
 			}
+
+			expLen := inSnap.State.RaftAppliedIndex - inSnap.State.TruncatedState.Index
+			if expLen != uint64(len(logEntries)) {
+				log.Fatal(ctx, "missing log entries in snapshot", expLen, len(logEntries))
```

Comment:
Instead of materializing the actual range of indexes received, and plumbing in HardState here (kvBatchSnapshotStrategy shouldn't know about that anyway), I'm warning loudly here instead. It will still Fatal and capture HardState + range of log indexes received when applying the snapshot.